src/chatgpt-api.ts

```diff
@@ -211,16 +211,17 @@ export class ChatGPTAPI {
                     result.id = response.id
                   }
 
-                  if (response?.choices?.length) {
+                  if (response.choices?.length) {
                     const delta = response.choices[0].delta
                     result.delta = delta.content
                     if (delta?.content) result.text += delta.content
-                    result.detail = response
 
                     if (delta.role) {
                       result.role = delta.role
                     }
 
+                    result.detail = response
```

Comment:
Same as with default fetch, `result.detail` should always exist in final message object